### ISSUE TITLE
version 3.9.9 update to 4.0.4 failed

### ISSUE THREAD

**Issue by mrainville** [bmlt-enabled/yap#590]
**Describe the bug**
Attempting to upgrade from 3.9.9 to 4.0.4

**What version are you running?**
Upgrading from version 3.9.9 to version 4.0.4

**Did this happen on prior versions to the one you are running?**
No.  HOWEVER WE ARE NOW DOWN and cannot use the older version 3.9.9

**To Reproduce**

- Downloaded https://github.com/bmlt-enabled/yap/releases/download/4.0.4/yap-4.0.4.zip
- uploaded the file to server /home3/arizonu9/public_html/yap
- extracted it to live-4.0.4
- logged into yap console with no problems at https://arizona-na.org/yap/live-4.0.4/admin/index.php
- Renamed old live directory to live-3.9.9
- Renamed live-4.0.4 to live
- attempted to connect with Twilio
- Received application error with Twilio  msg=HTTP+Connection+Failure+-+Read+timed+out&Msg=HTTP+Connection+Failure+-+Read+timed+out&sourceComponent=12000&ErrorCode=11205&url=https%3A%2F%2Farizona-na.org%2Fyap%2Flive%2Findex.php&LogLevel=ERROR

**Expected behavior**

**Screenshots**

**Additional context**

error_log file contents:
[15-May-2022 16:16:59 America/Boise] PHP Warning:  require_once(/home3/arizonu9/public_html/yap/live-4.0.4index.php): failed to open stream: No such file or directory in /home3/arizonu9/public_html/yap/live-4.0.4/server.php on line 21
[15-May-2022 16:16:59 America/Boise] PHP Fatal error:  require_once(): Failed opening required '/home3/arizonu9/public_html/yap/live-4.0.4index.php' (include_path='.:/opt/cpanel/ea-php74/root/usr/share/pear') in /home3/arizonu9/public_html/yap/live-4.0.4/server.php on line 21

**Comment by dgershman** [comment]
It looks like your webhook doesn't match the same thing as what the admin url is pointing to.

**Comment by dgershman** [comment]
This looks good and returns no error.  https://arizona-na.org/yap/live-4.0.4/index.php

**Comment by dgershman** [comment]
Change your webhook.  It looks like you have it set to https://arizona-na.org/yap/live/index.php instead of https://arizona-na.org/yap/live-4.0.4/index.php

**Comment by mrainville** [comment]
I had it reverted to version 3.9.9.  Instead of changing the webhooks for each number in Twilio.  I rename path..  Currently https://arizona-na.org/yap/live/index.php <https://arizona-na.org/yap/live/index.php> points to version 4.0.4 and https://arizona-na.org/yap/live-3.9.9/index.php <https://arizona-na.org/yap/live-3.9.9/index.php> is the old version. 

I suspect that I may have been experiencing problems with our hosting company while I was in the process of doing the upgrade; however the callouts are still failing and the reports are still timing out.

**Comment by dgershman** [comment]
Ok.  Well your ticket had inconsistent URLs and it seemed like an obvious issue at the time I looked.

From: Marc Rainville ***@***.***>
Date: Sunday, May 15, 2022 at 9:00 PM
To: bmlt-enabled/yap ***@***.***>
Cc: Danny Gershman ***@***.***>, Comment ***@***.***>
Subject: Re: [bmlt-enabled/yap] version 3.9.9 update to 4.0.4 failed (Issue #590)
I had it reverted to version 3.9.9. Instead of changing the webhooks for each number in Twilio. I rename path.. Currently https://arizona-na.org/yap/live/index.php <https://arizona-na.org/yap/live/index.php> points to version 4.0.4 and https://arizona-na.org/yap/live-3.9.9/index.php <https://arizona-na.org/yap/live-3.9.9/index.php> is the old version.

I suspect that I may have been experiencing problems with our hosting company while I was in the process of doing the upgrade; however the callouts are still failing and the reports are still timing out.

**Comment by dgershman** [comment]
If you'd like I can set up some time to go through this with you.

**Comment by mrainville** [comment]
I can get you into our server to look at it.  Sorry for the delayed response. Here's a little background on this issue.  Our hosting company was having issues while I was running the upgrade so I ended up restoring a backup and we're currently still on version 3.9.9.  I made a copy of the database and attempted the upgrade to a copy to see if I get the same error.  I haven't had a lot of time to get it working. I'll dig into it tonight and let you know what I find.

**Comment by mrainville** [comment]
[arizonu9_yaplive.sql.zip](https://github.com/bmlt-enabled/yap/files/8820339/arizonu9_yaplive.sql.zip)
I attempted to run the update to 4.0.4 tonight using a copy of this db at version 3.9.9.  The /admin/legacy/index.php starts the installer.php script which crashes with the following error in the /legacy/admin/error_log file:
if you'd like to get onto the server I can IM you the cpanel login info.
Thank you!
 
`[01-Jun-2022 20:26:01 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:30:58 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:31:40 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:44:47 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:46:34 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:48:56 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:54:31 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18
[01-Jun-2022 20:54:39 America/Boise] PHP Fatal error:  Uncaught Error: Call to undefined function url() in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php:18
Stack trace:
#0 {main}
  thrown in /home3/arizonu9/public_html/yap/yap-4.0.4/legacy/admin/installer.php on line 18`

**Comment by dgershman** [comment]
Ok the installer.PHP shouldn’t fire off if you copy over the config.PHP file.

Get Outlook for iOS<https://aka.ms/o0ukef>
________________________________

**Comment by dgershman** [comment]
Ok it looks good now
You sent
It looks like there may have not been a clean migration of the database when you rolled back.
You sent
https://arizona-na.org/yap/yap-4.0.4/ is pointed to the arizonu9_yap_v399 database.
[arizona-na.org](https://arizona-na.org/yap/yap-4.0.4/)
You sent
And it had the event_status table still,  I made a file in the root of the directory called .env and set APP_DEBUG=true which is how to see error messages in laravel.  https://laravel.com/docs/9.x/configuration#debug-mode

[Laravel - The PHP Framework For Web Artisans](https://laravel.com/docs/9.x/configuration#debug-mode)
You sent
Once i saw that, I could it trying to re-create the event_status table, which already existed.  I went into phpmyadmin and deleted the table, and hit reload in the web browser.
You sent
It created the table again, and now is able to continue forward.
